### DISCUSSION TITLE
Issues with electrode thickness/particle radius sensitivities

### DISCUSSION THREAD

**Discussion by AlvaroDFC** [pybamm-team/PyBaMM#4107] [Q&A]
Hello,

 I am using PyBamm 22.7 and I am trying to compute the sensitivities of voltage and temperature with respect to quantities that influence the scaling or the discretization, such as the "Negative electrode thickness" or "Negative particle radius" among others. This question is probably related to #2792.

I have been able to compute sensitivities w.r.t "Positive particle radius" and "Ambient temperature" simultaneously and it worked, yet when the sensitivities are computed w.r.t Negative particle radius" if prompts an error:

#############
pybamm.expression_tree.exceptions.SolverError: Could not find consistent states: Could not find acceptable solution: Error in Function::call for 'roots' [Newton] at .../casadi/core/function.cpp:361:
.../casadi/core/rootfinder.cpp:277: rootfinder process failed. Set 'error_on_fail' option to false to ignore this error.
#############

I have tried using the timescale option, but then I get a geometry error:

#############
ValueError: Geometry parameters must be Scalars after parameter processing
#############

I created a minimum working example to reproduce the issue.

```
import pybamm as pb
import numpy as np

# Creation of the models
# options = {"thermal": "lumped", "timescale": 1}  
options = {"thermal": "lumped"}                     
model = pb.lithium_ion.DFN(options=options)         
geometry = pb.battery_geometry(form_factor="cylindrical")
pb.set_logging_level("NOTICE")

# Time discretization
t_eval = np.arange(0, 400, 1)

# Definition of the chemistries
chemistry = pb.parameter_sets.Chen2020         
parameters = pb.ParameterValues(chemistry=chemistry)
parameters.update({"Negative electrode thickness [m]": "[input]",
                   "Negative particle radius [m]": "[input]"}) 
parameters.process_model(model)
parameters.process_geometry(geometry)

# Discretization and notice options
var_pts = {"x_n": 30, "x_s": 30, "x_p": 30, "r_n": 10, "r_p": 10}
mesh = pb.Mesh(geometry, model.default_submesh_types, var_pts)
discretization = pb.Discretisation(mesh, model.default_spatial_methods)
discretization.process_model(model)

# Solving the model
solver = pb.CasadiSolver(mode="fast")
solution = solver.solve(
    model, t_eval, inputs={'Negative electrode thickness [m]': 8.52e-05,
                           'Negative particle radius [m]': 2.5e-06},
    calculate_sensitivities=True
)

# solution vectors
voltage = solution["Terminal voltage [V]"].entries
dVdLn   = solution["Terminal voltage [V]"].sensitivities['Negative electrode thickness [m]']
dVdrn   = solution["Terminal voltage [V]"].sensitivities['Negative particle radius [m]']
````

Thank you!

**Comment by martinjrobins** [comment]
Hi @AlvaroDFC, you shouldn't be able to set any geometry parameters to be input parameters (and therefore you won't be able to calculate their sensitivities this way, although you could do it manually using finite differences). When I run your script on the `develop` branch I get:

```
/home/mrobins/git/PyBaMM/script.py:15: DeprecationWarning: Parameter sets should be called directly by their name (Chen2020), instead of via pybamm.parameter_sets (pybamm.parameter_sets.Chen2020).
  chemistry = pb.parameter_sets.Chen2020
Traceback (most recent call last):
  File "/home/mrobins/git/PyBaMM/script.py", line 20, in <module>
    parameters.process_geometry(geometry)
  File "/home/mrobins/git/PyBaMM/pybamm/parameters/parameter_values.py", line 597, in process_geometry
    geometry[domain][spatial_variable][lim] = process_and_check(sym)
  File "/home/mrobins/git/PyBaMM/pybamm/parameters/parameter_values.py", line 581, in process_and_check
    raise ValueError(
ValueError: Geometry parameters must be Scalars after parameter processing
(env) ➜  PyBaMM git:(develop) ✗ 
```

**Comment by AlvaroDFC** [comment]
@martinjrobins thank you for your answer. I will carry out the finite differencing step although it is not ideal for my problem. I think it would be a great feature to include the geometry parameters in the AD capability.
